tests/filters_test.go

```diff
@@ -0,0 +1,170 @@
+// Copyright 2021 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/apigee/registry/connection"
+	"github.com/apigee/registry/gapic"
+	"github.com/apigee/registry/rpc"
+	"google.golang.org/api/iterator"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func unavailable(err error) bool {
+	if err == nil {
+		return false
+	}
+	st, ok := status.FromError(err)
+	if !ok {
+		return false
+	}
+	return st.Code() == codes.Unavailable
+}
+
+func check(t *testing.T, message string, err error) {
```

Comment:
Can this be checked in-line?

The check for unavailable doesn't seem to add much value compared to the actual error: `rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp [::1]:8080: connect: connection refused"`. If desired we could do one availability check at the start of the tests to make it explicitly clear that a local server should be running, but I think the message is quite clear.

Assertion helpers like these are used like domain-specific languages for test assertions, with drawbacks such as the indirection preventing linters from telling us if the message string we provide isn't valid. This also makes errors look like they occur at line 47 regardless of where the check was called (this can be fixed though).

Reply:
This was reused from elsewhere - I think it would make sense to tighten this up and put it somewhere common, how about in a separate PR focused on test architecture?